internal/connector/test/integration/features/connector-agent-api.feature

```diff
@@ -584,11 +584,70 @@ Feature: connector agent API
     Then the response code should be 200
     And the ".status.state" selection from the response should match "ready"
 
+    #-----------------------------------------------------------------------------------------------------------------
+    # In this part of the Scenario we test whether agent gets deployments even when missing secrets in the vault
+    #-----------------------------------------------------------------------------------------------------------------
+    # save the service account secret in the client id and set the secret to an invalid value
+    Given I run SQL "UPDATE connectors SET service_account_client_id=service_account_client_secret, service_account_client_secret='missing-secret' WHERE id = '${connector_id}';" expect 1 row to be affected.
+    # check that the deployment update is still sent to the agent without a spec and phase set to 'secrets-unavailable'
+    Given I am logged in as "Shard"
+    Given I set the "Authorization" header to "Bearer ${shard_token}"
+    And I wait up to "5" seconds for a response event
+    Then the ".object.spec.connector_spec" selection from the response should match "null"
```

Comment:
@lburgazzoli note that the agent api is setting the spec to `{}`. But that gets marshaled to `nil` in it's json handling so `connector_spec` will be missing in the deployment.